mitodl/mitxonline
# Update react-redux and related packages to facilitate catalog changes

# What are the relevant tickets?
Supports https://github.com/mitodl/hq/issues/2992

# Description (What does it do?)
Upgrades:
react-redux: 5.1.2->7.1.0
redux: 3.7.2->4.0.0
redux-query: 2.3.1->3.3.1 at which point it split some things out into also:
redux-query-interface-superagent (literally the same, just in a new spot in their v3)
redux-query-react (same as the superagent)
Added some test dependencies top-level

Keeps shallow rendering for many tests - thank you to Chris for the help!
Moved a few over to use full mount(), but because it's using a different renderer, it's not behaving on requests. I'm going to attempt to mock out the responses one more time, but I'm posting this to see if anyone else has another/ a better idea.

I've added a few comments as you go along to help explain a little of this - but there's a lot.

# How can this be tested?
First - run the tests and everything - be sure they pass.
Check the tests that were changed to be sure the changes make sense.
Lastly, Check some of the processes/workflows:
- registration especially - this is impacted by the changes and pretty important
- catalog bits
- anything that pops up a notification
- look for errors, weird behavior, etc.
